src/cmd/actions.rs

```diff
@@ -0,0 +1,63 @@
+/// Represents actions that are available while reading live output from a process
+pub struct Actions {
+    lines: Vec<String>,
+}
+
+impl<'a> Actions {
+    pub(crate) fn new() -> Self {
+        Actions { lines: Vec::new() }
+    }
+
+    pub(crate) fn next_input(&mut self, input: &str) {
```

Comment:
You can accept a `String` here, to avoid a `String -> &str -> String` conversion. I'm pretty sure the original line is not used anywhere after you pass it to this method.